Feature/webapi

YamiTheCreator opened the pull request.

YamiTheCreator committed:
Delete Fighters directory

YamiTheCreator committed:
Delete OrderManager directory

YamiTheCreator committed:
Complete WebApi Task

yaroslav-hub commented on WebApi/Domain/Entities/Property.cs:
```diff
@@ -0,0 +1,13 @@
+namespace Domain.Entities;
+public class Property
+{
+    public Guid Id { get; set; }
```
Зачем Id Guid? Почему не int?

yaroslav-hub commented on WebApi/Domain/Repositories/IPropertyRepository.cs:
```diff
@@ -0,0 +1,12 @@
+
+public interface IPropertyRepository
+{
+    Task<IEnumerable<Property>> GetAllAsync();
```
Зачем везде Async? Перепиши на синхронную работу с контроллеров до репозиториев

yaroslav-hub commented on WebApi/Domain/Repositories/IPropertyRepository.cs:
```diff
@@ -0,0 +1,12 @@
+public interface IPropertyRepository
+{
+    Task<IEnumerable<Property>> GetAllAsync();
+    Task<Property?> GetByIdAsync(Guid id);
```
Отсутствует форматирование, соответствующее .editorconfig

yaroslav-hub commented on WebApi/Infrastructure/Repositories/ReservationRepository.cs:
```diff
@@ -0,0 +1,75 @@
+        DateTime? endDate,
+        string? guestName)
+    {
+        var query = _context.Reservations.AsQueryable();
```
var не используем

yaroslav-hub commented on WebApi/Infrastructure/Repositories/ReservationRepository.cs:
```diff
@@ -0,0 +1,75 @@
+    public async Task<bool> UpdateAsync(Reservation reservation)
+    {
+        var existing = await _context.Reservations.FindAsync(reservation.Id);
+        if (existing == null)
```
На null лучше проверять через is и is not

yaroslav-hub commented on WebApi/Infrastructure/Repositories/ReservationRepository.cs:
```diff
@@ -0,0 +1,75 @@
+        
+        if (startDate.HasValue && endDate.HasValue)
+            query = query.Where(r => 
+                r.ArrivalDate < endDate && 
```
Тут включительно должно быть, отрезок

yaroslav-hub commented on WebApi/Infrastructure/Services/RoomTypeService.cs:
```diff
@@ -0,0 +1,67 @@
+        if (roomType.MinPersonCount > roomType.MaxPersonCount)
+            throw new ArgumentException("MinPersonCount cannot be greater than MaxPersonCount.");
+
+        roomType.Id = Guid.NewGuid();
```
По хорошему Id должны быть числовыми и генерироваться автоматически, а не задаваться вручную. В данной лабораторной Guid не оправдан

yaroslav-hub commented on WebApi/WebApi/Controllers/PropertyController.cs:
```diff
@@ -0,0 +1,128 @@
+    [HttpPost]
+    public async Task<IActionResult> Create([FromBody] AddPropertyContract contract)
+    {
+        if (!ModelState.IsValid)
```
Для читаемости лучше явно обозначать блоки кода в фигурные скобки

yaroslav-hub commented on WebApi/WebApi/Program.cs:
```diff
@@ -0,0 +1,150 @@
+app.Run();
+return;
+
+static void SeedData(ApplicationDbContext context)
```
Необходимо добавить миграции, написать конфигурации и подключить к бд MS SQL. SeedData можно оставить, но заполнить уже бд, а не контекст на уровне приложения

yaroslav-hub commented on WebApi/WebApi/Controllers/SearchController.cs:
```diff
@@ -0,0 +1,88 @@
+
+        try
+        {
+            var properties = await _propertyRepository.GetAllAsync();
```
Логику поиска необходимо вынести в отдельный сервис

YamiTheCreator committed:
Some changes

Вынес search в отдельный сервис и переделал методы под синхронную работу. По заданию можно использовать guid и inMemory реализацию, поэтому это менять не стал. применил editorconfig, слегка поправил логику и маршруты. заменил var на explicit типы
